arches_controlled_lists/src/arches_controlled_lists/widgets/ReferenceSelectWidget/components/ReferenceSelectWidgetEditor.vue

```diff
@@ -117,15 +88,29 @@ async function getOptions() {
 
 // let timeout: ReturnType<typeof setTimeout>;
 
-function resolver(e: FormFieldResolverOptions) {
-    validate(e);
+function resolver({ values }: FormFieldResolverOptions) {
+    validate(values);
     // return new Promise((resolve) => {
     //     if (timeout) clearTimeout(timeout);
 
     //     timeout = setTimeout(() => {
     //         resolve(validate(e));
     //     }, 500);
     // });
+    const nodeAlias = props.nodeAlias;
+    let selectedItemKeys: string[] = [];
+    if (values) {
+        selectedItemKeys = Object.entries(values).reduce<string[]>(
+            (keys, [key, val]) => {
+                if (val === true) keys.push(key);
+                return keys;
+            },
+            [],
+        );
+    }
+    return {
```

Comment:
I'm not sure this is working as intended. I attempt to minimally repro with the NonLocalizedStringWidget via

```
function resolver({values}: FormFieldResolverOptions) {
    // validate(e);
    return 'foo';
}
```

However, the value saved to the form is what I've input into field, not `foo`.


I understand the `ReferenceSelectWidget` is not the `NonLocalizedStringWidget`, but I'd expect any transform pattern to be usable in all widgets.

Reply:
for documentation purposes: resolvers return a dict with `values` and `errors` as top level keys, so you need to patch the updated node value into the values object like this:
```
function resolver({values}: FormFieldResolverOptions) {
  return {
          values: {
              [nodeAlias]: 'BAZ',
          }
      };
```